src/main/java/seedu/address/model/application/Date.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.model.application;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * Represents the Date that the user applied to the Position of the Company.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Date {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in the format of dd/mm/yyyy, dd-mm-yyyy or dd.mm.yyyy with proper month "
+                    + "and leap year support";
+    //Only valid if dd/mm/yyyy, dd-mm-yyyy or dd.mm.yyyy with month and leap year support
+    //Retrieve from https://stackoverflow.com/questions/15491894/regex-to-validate-date-formats-dd-mm-yyyy-dd-mm-yyyy
+    // -dd-mm-yyyy-dd-mmm-yyyy
+    public static final String VALIDATION_REGEX = "^(?:(?:31(\\/|-|\\.)(?:0?[13578]|1[02]))\\1|(?:(?:29|30)"
+            + "(\\/|-|\\.)(?:0?[13-9]|1[0-2])\\2))(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$|^(?:29(\\/|-|\\.)0?2\\3(?:("
+            + "?:(?:1[6-9]|[2-9]\\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))"
+            + "$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9])|(?:1[0-2]))\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2"
+            + "})$";
+    public final LocalDate value;
+
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param dateString A valid date in String.
+     */
+    public Date(String dateString) {
+        requireNonNull(dateString);
+        checkArgument(isValidDate(dateString), MESSAGE_CONSTRAINTS);
+        value = this.parseLocalDate(dateString);
+    }
+
+    /**
+     * Returns true if a given string is a valid date string.
+     */
+    public static boolean isValidDate(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    private LocalDate parseLocalDate(String dateString) {
+        return LocalDate.parse(dateString);
```

Comment:
Does it work? I remember `LocalDate.parse` by default only works with string in the format of yyyy-mm-dd. I think you can specify the format you want like this.
```suggestion
        return LocalDate.parse(dateString, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```
It's hard to allow for 3 formats though.

Reply:
Indeed, thanks for the suggestion. I narrow down the possible inputs to just yyyy-MM-dd.